onnxruntime/core/graph/graph.cc

```diff
@@ -3030,6 +3030,96 @@ ONNX_NAMESPACE::GraphProto Graph::ToGraphProto() const {
   return result;
 }
 
+ONNX_NAMESPACE::GraphProto Graph::ToGraphProtoWithExternalInitializers(const std::string& external_file_name) const {
+  // Sparse tensors are not supported yet.
```

Comment:
Could this instead be an enhancement to the const Graph::ToGraphProto method so that things like sparse tensors get handled consistently, and ToGraphProtoInternal is used instead of duplicating that logic here?

And regarding sparse tensors, why not just store them in the graph proto if you don't want to write them to the external file?

Reply:
Thanks, I have reworked this function to follow your suggestion, indeed it looks better now.